### PR TITLE
low_voltage tools migration 

### PR DESCRIPTION
**Pull request by cdozen**
low_voltage migration : 
- no "unittest" yet!

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on:
```
Comment (cbernet) [inline .github/workflows/python-package.yml:6]: please take the web server as an example. we miss black, tests, coverage

File: docker/tedd_low_voltage/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ports:
+      - "1883:1883"
+    expose:
+      - "1883"
+    volumes:
+      - mosquitto_data:/mosquitto/data
+      - mosquitto_log:/mosquitto/log
+      - ./mosquitto/config:/mosquitto/config
+    
+  lv_hmp4040:
```
Comment (cbernet) [inline docker/tedd_low_voltage/docker-compose.yml]: shouldn't be in test stack. we only add supporting services there, see web server

File: low_voltage/lv-HMP/HMP4040.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+#for python2 
+#import visa
+#rm = visa.ResourceManager("/usr/lib64/librsvisa.so@ni")
+#for python3
+import pyvisa as visa
+rm = visa.ResourceManager("/usr/lib64/librsvisa.so@ivi")
```
Comment (cbernet) [inline low_voltage/lv-HMP/HMP4040.py:7]: black has not been run

File: low_voltage/lv-HMP/hmp.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        idn = self.HMP4040.read()
+        print("IDN:", idn) 
+        self.HMP4040.set_visa_attribute(visa.constants.VI_ATTR_TERMCHAR_EN,True)
+        attr = self.HMP4040.get_visa_attribute(visa.constants.VI_ATTR_TERMCHAR_EN)
+        print("Attrib. TERMCHAR_EN:", attr)
+        self.HMP4040.set_visa_attribute(visa.constants.VI_ATTR_SUPPRESS_END_EN,False)
+        attr = self.HMP4040.get_visa_attribute(visa.constants.VI_ATTR_SUPPRESS_END_EN)
+        print("Attrib. SUPPRESS_END_EN:", attr)
+        print()
+    def command(self, topic: str, message: Union[bytes, float, int]) -> None:
```
Comment (cbernet) [inline low_voltage/lv-HMP/hmp.py:51]: black

File: low_voltage/lv-HMP/hmp.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    vreq: float = 0.
+    #curr: float =0.
+
+
+class HMP(object):
+    rm = visa.ResourceManager("/usr/lib64/librsvisa.so@ivi")
+    HMP4040 = rm.open_resource('TCPIP::192.168.1.202::10002::SOCKET')
+    num_of_channels = 5
+
+    def __init__(self, name: str, n_channels: int = 5):
```
Comment (cbernet) [inline low_voltage/lv-HMP/hmp.py:28]: your methods must be  commented

File: low_voltage/mqtt.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import os
```
Comment (cbernet) [inline low_voltage/mqtt.py:1]: this file must be unittested. 
i understand hmp.py cannot, but you can mock it.

File: requirements.txt [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (cbernet) [inline requirements.txt:2]: make sure the requirements are the same as in web server when applicable

File: requirements/local.txt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+-r ./base.txt
```
Comment (cbernet) [inline requirements/local.txt:1]: same

**Review by cbernet — CHANGES_REQUESTED** [review]
Hi @cdozen nice, here are my comments. 
And please add a detailed README !

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (cdozen) [reply .github/workflows/python-package.yml:6]: Hi, I added the black, pytest, and coverage into the python-package.yml. But coverage failed. I don't know how to solve it.

File: requirements.txt (hunk 7 above)
Comment (cdozen) [reply requirements.txt:2]: I used the requirements folder (copied from the webserver ) in dockerfile.

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (cbernet) [reply .github/workflows/python-package.yml:6]: should be fine now

File: docker/tedd_low_voltage/docker-compose.yml (hunk 2 above)
Comment (cbernet) [reply docker/tedd_low_voltage/docker-compose.yml]: done

File: low_voltage/lv-HMP/HMP4040.py (hunk 3 above)
Comment (cbernet) [reply low_voltage/lv-HMP/HMP4040.py:7]: done

File: low_voltage/lv-HMP/hmp.py (hunk 5 above)
Comment (cdozen) [reply low_voltage/lv-HMP/hmp.py:28]: added some comments

**Comment by cdozen** [comment]
> 

readme added

File: README.md [hunk 9]
```diff
@@ -1,2 +1,43 @@
...
+conda activate low_voltage
+python3 -m pip install --upgrade pip
+python3 -m pip install -r requirements/local.txt
+python3 -m pip install -e .  
+```
+## Dockerfile
+To create centos:low_votage image with necessary packages.
+
+```
+docker build -t centos:tedd_dcs_low_voltage .
```
Comment (cbernet) [inline README.md:19]: tedd_dcs_low_voltage_centos

Comment (cbernet) [reply README.md:19]: and update readme accordingly

**Review by cbernet — APPROVED** [review]

**Pull request merged**